### PR TITLE
Must use RFC 3339 date and time formats

### PR DESCRIPTION
**Pull request by sniff**

### CODE REVIEW

File: chapters/json-guidelines.adoc [hunk 1]
```diff
@@ -122,27 +122,18 @@ Strings are a reasonable target for values that are by design enumerations.
 
 
 [#126]
-== {SHOULD} Date property values should conform to RFC 3339
+== {MUST} Date property values must conform to RFC 3339
 
-Use the date and time formats defined by {RFC-3339}#section-5.6[RFC 3339]:
+Use only the date and time formats defined by {RFC-3339}#section-5.6[RFC 3339]:
```
Comment (apankowski) [inline chapters/json-guidelines.adoc]: I remember that during the discussion we mentioned that we would simply stick to the ISO 8601 standard and refer to format names used there. Or did you maybe found that the RFC-3339 is better? Either way I'd add "time" representation which is missing here (to me) & update title accordingly. Something like:
```
== {MUST} Date & time property values must conform to ISO 8601

Use only the date, time and date-time formats defined by {ISO-8601}[ISO 8601] standard:
* "date" values should match "date" format of the standard;
* "time" values should match "time" format of the standard;
* "date-time" values should match "combined date and time" format of the standard.
```
what do you (and others) think?

Comment (sniff) [reply chapters/json-guidelines.adoc]: I found that RFC-3339 section 5.6 is much easy to read and understand then ISO. First of all, you have to pay to access the ISO :) also RFC-3339 is a subset of ISO.
Not sure if we should use "time" without relation to the date, that's why I prefer to skip it here.
I've updated the guideline to be more specific.

Comment (apankowski) [reply chapters/json-guidelines.adoc]: thanks for update on this - that's a valid point. regarding time-only properties - as far as I know financial service team already mentioned that there are external systems that differ in what they provide as e.g. invoice date - some provide only date, other date-time. it seems that the connectors will have to split it to date + time (separate properties), that's why I mentioned it.

File: chapters/json-guidelines.adoc [hunk 2]
```diff
@@ -120,27 +120,16 @@ Strings are a reasonable target for values that are by design enumerations.
...
-Use the date and time formats defined by {RFC-3339}#section-5.6[RFC 3339]:
+For representation of date and time use only the following formats defined
+by {RFC-3339}#section-5.6[RFC 3339]:
 
-* for "date" use strings matching
-`date-fullyear "-" date-month "-" date-mday`, for example: `2015-05-28`
-* for "date-time" use strings matching `full-date "T" full-time`, for
-example `2015-05-28T14:07:17Z`
+* for "date" use strings matching RFC `full-date` format, example: `2015-05-28`.
+* for "date and time" use strings matching RFC `date-time` format with only UTC zone offset, example `2015-05-28T14:07:17Z`.
```
Comment (wilerson) [inline chapters/json-guidelines.adoc]: Suggestion: add a "time of day" option, matching RFC `partial-time` format (and proposed OpenAPI `time` format, as in https://github.com/OAI/OpenAPI-Specification/issues/845#issuecomment-378139730 ).

File: chapters/json-guidelines.adoc [hunk 3]
```diff
@@ -155,6 +144,12 @@ timestamps, but this can introduce interpretation issues with precision
 1460062925000 or 1460062925.000. Date strings, though more verbose and
 requiring more effort to parse, avoid this ambiguity.
 
+Note that the
+https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types[OpenAPI
```
Comment (wilerson) [inline chapters/json-guidelines.adoc]: Let's try linking to OpenAPI v3 whenever possible: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#data-types

Comment (sniff) [reply chapters/json-guidelines.adoc]: Updated across the whole project in the separate PR
https://github.com/infinitecsolutions/restful-api-guidelines/pull/18

**Pull request merged**